File: solutions/python/collatz-conjecture/collatz_conjecture.py
def steps(num: int) -> int:
    """Collatz Conjecture"""
    if num < 1:
        raise ValueError("Only positive integers are allowed")
    count: int = 0
    while num > 1:
        count += 1
        num = 3 * num + 1 if num % 2 else num // 2
    return count

File: solutions/cpp/collatz-conjecture/collatz_conjecture.cpp
#include "collatz_conjecture.h"
#include <stdexcept>

namespace collatz_conjecture
{
    unsigned int steps(int num)
    {
        if (num < 1)
            throw std::domain_error("Only positive numbers are allowed");

        unsigned int steps{0};
        for (; num > 1; steps++)
            num = (num % 2 == 0) ? num / 2 : 3 * num + 1;
        return steps;
    }

} // namespace collatz_conjecture
